semantic-analysis/src/Analysis/Language.hs

```diff
@@ -0,0 +1,136 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DeriveAnyClass #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE KindSignatures #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE OverloadedStrings #-}
+module Analysis.Language
```

Comment:
@tclem pointed out that this makes diffing, parsing, etc. have a dependency on `semantic-analysis` when we could get away with having it inside `semantic` alone. I could go either way on this; thoughts?

Reply:
IMO this file belongs in `semantic-source`, altho that is a bit more work at the moment due to us getting that package from hackage. (cf #423)

Reply:
I’d wait until #423 to fix this, yeah.

Reply:
I don’t think we should wait; the churn this introduces for the other packages makes it worth the extra effort for the `semantic-source` release, IMO.

If it helps, we could potentially roll this in with @aymannadeem’s intended changes to `semantic-source` to only do a single release; I’ll leave that to the two of you to suss out, tho.

Reply:
(Note to self: I’m also wondering if `Analysis.File` doesn’t belong in `semantic-source`, but that is orthogonal and should not hold up this PR.)